inc/ticket.class.php

```diff
@@ -6422,6 +6425,82 @@ static function cronCreateInquest($task) {
    }
 
 
+   /**
+    * Cron for ticket's automatic purge
+    *
+    * @param Crontask $task Crontask object
+    *
+    * @return integer (0 : nothing done - 1 : done)
+   **/
+   static function cronPurgeTicket(Crontask $task) {
+      global $DB;
+
+      $ticket = new self();
+
+      //search entities
+      $tot = 0;
+
+      $entities = $DB->request(
+         [
+            'SELECT' => 'id',
+            'FROM'   => Entity::getTable(),
+         ]
+      );
+
+      foreach ($entities as $entity) {
+         $delay  = Entity::getUsedConfig('autopurge_delay', $entity['id'], '', Entity::CONFIG_NEVER);
+         if ($delay >= 0) {
+            $criteria = [
+               'FROM'   => $ticket->getTable(),
+               'WHERE'  => [
+                  'entities_id'  => $entity['id'],
+                  'status'       => $ticket->getClosedStatusArray(),
+               ]
+            ];
+
+            if ($delay > 0) {
+               $calendars_id = Entity::getUsedConfig('calendars_id', $entity['id']);
+               if ($calendars_id > 0) {
+                  $calendar = new Calendar;
+                  $calendar->getFromDB($calendars_id);
+                  $end_date = $calendar->computeEndDate(
+                     date('Y-m-d H:i:s'),
+                     - $delay * DAY_TIMESTAMP,
+                     0,
+                     true
+                  );
+                  $criteria['WHERE']['closedate'] = ['<=', $end_date];
+               } else {
+                  // no calendar, remove all days
+                  $criteria['WHERE']['closedate'] = new \QueryExpression("ADDDATE(`closedate`, INTERVAL ".$delay." DAY) < NOW()");
+               }
+            }
+
+            $iterator = $DB->request($criteria);
+            $nb = 0;
+
+            foreach ($iterator as $tick) {
+               $ticket->delete(
```

Comment:
> automatic action to purge closed tickets **and their attached documents**

It misses attached documents (DB item + storage file) purge, no?

Reply:
I think this is done along with ticket deletion, I'll check

Reply:
As discussed, I'm going to add another crontask in order to clean orphaned documents